src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/NuGetWorkspaceCommandHandler.cs

```diff
@@ -65,24 +79,64 @@ public bool QueryStatus(List<WorkspaceVisualNodeBase> selection, Guid pguidCmdGr
             if (pguidCmdGroup == CommandGroup.NuGetOnlineEnvironmentsClientProjectCommandSetGuid)
             {
                 var nCmdIDInt = (int)nCmdID;
-
-                if (IsSolutionOnlySelection(selection))
                 {
                     switch (nCmdIDInt)
                     {
                         case PkgCmdIDList.CmdidRestorePackages:
-                            var isRestoreActionInProgress = _restoreCommandHandler.IsRestoreActionInProgress();
-                            cmdf = (uint)((isRestoreActionInProgress ? 0 : Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED)
-                                | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED);
-                            handled = true;
+                            if (IsSolutionOnlySelection(selection))
+                            {
+
+                                var isRestoreActionInProgress = _restoreCommandHandler.IsRestoreActionInProgress();
+                                cmdf = (uint)((isRestoreActionInProgress ? 0 : Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED)
+                                    | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED);
+                                handled = true;
+                            }
+                            break;
+                        case PkgCmdIDList.CmdIdManageProjectUI:
+                            if (IsSupportedProjectSelection(selection))
+                            {
+                                var isPackageManagerUISupported = _packageManagerUICommandHandler.IsPackageManagerUISupported(selection.Single());
+                                cmdf = (uint)(isPackageManagerUISupported ?
+                                    (Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED) :
+                                    Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_INVISIBLE);
+                                handled = true;
+                            }
+                            break;
+                        default:
                             break;
                     }
                 }
             }
-
             return handled;
         }
 
+        private static bool IsSupportedProjectSelection(List<WorkspaceVisualNodeBase> selection)
+        {
+            if (selection != null &&
+                selection.Count.Equals(1))
+            {
+                // We support every item representing a project
+                // - we don't have the ability to do a capabilities check, because we don't have enough information.
+                string fileExtension;
+                try
+                {
+                    fileExtension = Path.GetExtension(selection.Single().NodeMoniker);
+                }
+                catch (ArgumentException)
+                {
+                    return false;
```

Comment:
Can we log the exception then return `false` instead of suppressing?

Reply:
What would we do with the exception? 
I have an issue to think through that: https://github.com/NuGet/Home/issues/9352